src/main/java/org/kohsuke/github/PagedIterator.java

```diff
@@ -38,29 +42,23 @@
 
     public boolean hasNext() {
         fetch();
-        return current != null;
+        return current.length > pos;
     }
 
     public T next() {
-        fetch();
-        if (current == null)
+        if (!hasNext())
             throw new NoSuchElementException();
         return current[pos++];
     }
 
     private void fetch() {
-        while (current == null || current.length <= pos) {
-            if (!base.hasNext()) {// no more to retrieve
-                current = null;
-                pos = 0;
-                return;
-            }
-
-            current = base.next();
-            wrapUp(current);
+        if ((current == null || current.length <= pos) && base.hasNext()) {
```

Comment:
Not really specific to this PR, but, I could use a quick overview of the relationships between `current`, `pos`, and `base`. I thought I had it figured out for a minute but I guess I don't.

Reply:
I can pick better names for these and give them a little more commenting.